docs/reference/developer/notebook-style-guide.ipynb

```diff
@@ -0,0 +1,78 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "3205d1c4-8704-4b77-80b6-287e252864eb",
+   "metadata": {},
+   "source": [
+    "# Jupyter Notebook Style Guide\n",
+    "\n",
+    "## Motivation\n",
+    "\n",
+    "This style guide is meant for anyone writing documentation pages, tutorials, or workflows as Jupyter notebooks.\n",
+    "\n",
+    "The purpose is to make notebooks easy to understand and readable, and to minimize the reader's cognitive overload when switching between reading various notebooks.\n",
+    "\n",
+    "## Recommendations\n",
+    "\n",
+    "### Saving notebooks\n",
+    "\n",
+    "- Avoid storing outputs before saving or committing a notebook.\n",
+    "  - Use \"Restart kernel and clear all outputs\" before saving.\n",
+    "  - [nb-clean](https://github.com/srstevenson/nb-clean) can be used to do this automatically.\n",
+    "\n",
+    "### Document structure\n",
+    "\n",
+    "- Use markdown sections, subsections, ... for easy navigation and readability.\n",
+    "    \n",
+    "### Portability\n",
+    "\n",
+    "- Avoid hard-coding file paths which would require users to download files into specific folders.\n",
+    "- Use APIs, not file systems, to access data.\n",
+    "  - [pooch](https://www.fatiando.org/pooch/latest/) is a good candidate when sample files are needed.\n",
```

Comment:
Should we also link to an example where we are doing this in Scipp?
Because it took you a bit of time to figure out how best to use `pooch`, and this would be helpful in addition to linking to the main pooch docs pages.

Reply:
Not sure I want to claim that we use pooch in the best/canonical way. The documentation is good so I would prefer to just link to that.

Also, we don't use pooch in scipp itself.